saito-spammer-new/src/saito/transaction_generator.rs

```diff
@@ -53,23 +53,29 @@ impl TransactionGenerator {
         };
     }
 
-    pub async fn on_new_block(&mut self) -> Option<Vec<Transaction>> {
+    pub fn get_state(&self) -> GeneratorState {
+        return self.state.clone();
+    }
+    pub async fn on_new_block(&mut self) -> Option<LinkedList<Transaction>> {
+        let mut transactions: Option<LinkedList<Transaction>> = None;
+
         match self.state {
             GeneratorState::CreatingSlips => {
-                return self.create_slips().await;
+                transactions = self.create_slips().await;
             }
             GeneratorState::WaitingForBlockChainConfirmation => {
                 if self.check_blockchain_for_confirmation().await {
-                    return self.create_test_transactions().await;
+                    transactions = self.create_test_transactions().await;
+                    self.state = GeneratorState::Done;
                 }
             }
-            GeneratorState::Ready => {}
+            GeneratorState::Done => {}
         }
 
-        return None;
+        return transactions;
     }
 
-    async fn create_slips(&mut self) -> Option<Vec<Transaction>> {
+    async fn create_slips(&mut self) -> Option<LinkedList<Transaction>> {
```

Comment:
VecDeque ?

Reply:
Changed it to a list because the logic needs a list, might work with a VecDeque but I don't think it will out perform list in this push_back pop_front FIFO configuration.